mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/MlflowAutologEventPublisher.scala

```diff
@@ -149,12 +149,17 @@ private[autologging] trait MlflowAutologEventPublisherImpl {
   private val checkpointFilePattern = ".*\\d+\\.checkpoint(\\.\\d+\\.\\d+)?\\.parquet$".r.pattern
   private def isCheckpointFile(path: String): Boolean = checkpointFilePattern.matcher(path).matches()
 
+  private def shouldSkipPublish(path: String, format: Option[String]): Boolean = {
+    // 1. Spark first loads head of the data as unknown "text" to infer the schema, which we don't want to log
+    // 2. Checkpoint files don't provide useful information, so we filter them out
+    (format.isEmpty || format.get == "text") || isCheckpointFile(path)
+  }
+
```

Comment:
/ai Is `format.forall(f => f == "text")` equivalent to `(format.isEmpty || format.get == "text")`?